src/main/java/com/exadel/sober/repositoriesImplemetation/PromiseRepositoryImpl.java

```diff
@@ -0,0 +1,56 @@
+package com.exadel.sober.repositoriesImplemetation;
+
+import com.exadel.sober.models.Promise;
+import com.exadel.sober.models.Reason;
+import com.exadel.sober.repositories.PromiseRepository;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public class PromiseRepositoryImpl implements PromiseRepository {
+    private final JdbcTemplate jdbcTemplate;
+
+    public PromiseRepositoryImpl(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+
+    @Override
+    public void save(Promise promise) {
+        jdbcTemplate.update("insert promise (addiction_id, user_id, expired_date) values (?, ?, ?)",
+                promise.getAddictionId(),
+                promise.getUserId(),
+                promise.getExpiredDate());
+    }
+
+    @Override
+    public Promise findPromiseByPromiseId(Integer promiseId) {
+        return (Promise) jdbcTemplate.queryForObject("select * from promise where promise_id = ?",
```

Comment:
seems you not need cast to promise there. Remove (Promise)

Reply:
we discussed it on call - without cast need custom mapper